src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax;
+
+import core.basesyntax.db.FruitShopStorage;
+import core.basesyntax.db.ShopStorage;
+import core.basesyntax.model.AbstractTransaction;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.service.CsvReaderService;
+import core.basesyntax.service.ReportGenerationService;
+import core.basesyntax.service.ShopCalculationService;
+import core.basesyntax.service.impl.CsvReaderServiceImpl;
+import core.basesyntax.service.impl.FruitReportGenerationServiceImpl;
+import core.basesyntax.service.impl.FruitShopCalculationServiceImpl;
+import java.util.List;
+
+public class Main {
+    public static void main(String[] args) {
+        CsvReaderService<Fruit> csvReaderService = new CsvReaderServiceImpl();
+        List<AbstractTransaction<Fruit>> fruitsOperations = csvReaderService.parse(
+                "src" + "/main/resources/file.csv");
+        ShopCalculationService<Fruit> shopCalculationService = new FruitShopCalculationServiceImpl(
+                new FruitShopStorage());
+        ShopStorage<Fruit> fruitStorage = shopCalculationService.calculate(fruitsOperations);
```

Comment:
The `calculate` method in `FruitShopCalculationServiceImpl` should return a `ShopStorage<T>` instead of `FruitShopStorage` to maintain consistency with the generic type `T` in the `ShopCalculationService` interface. This change will enhance flexibility and align with the interface definition.